### ISSUE TITLE
Z-link or three ways-connection

### ISSUE THREAD

**Issue by serrq** [simplex-chat/simplex-chat#1502]
I found (theorically) a new way of connecting two people (unlinked) using a mutual contact.

We know for sure that friend in common is a secure transmission channel because it is covered by encryption and the SimpleX protocol, so to make a z-link we make the connection request to contact B (for example my mother, already present among my contacts) which forwards it to C (my sister and target of the operation). C accepts my request conveyed through B and communicates it to me (A) again exploiting B as a transmission vector (the reverse path).

B just has to select which contact is the recipient of that z-link, the rest takes place all automatically.

What problem solves this solution?

The problem of having physically in presence C (my sister lives away from me), and exchanging a pairing link without using external channels, typically insecure.

**Comment by epoberezkin** [comment]
Yes, we called this theoretic feature an "introduction". We decided it's a [relatively] low value as it can be done manually – B can send then link to A, and A would forward it to C (via copy/paste).

But yes, it's a consideration for the future.
